blueflood-core/src/main/java/com/rackspacecloud/blueflood/inputs/formats/JSONMetricsContainer.java

```diff
@@ -142,6 +147,13 @@ public void setTtlInSeconds(int ttlInSeconds) {
             this.ttlInSeconds = ttlInSeconds;
         }
 
+        public boolean isValid() {
+            long current = System.currentTimeMillis();
+
+            // valid if collectionTime is greater than past window or less than future window
+            return ((collectionTime >= current - pastDiff ) && ( collectionTime <= current + futureDiff ));
```

Comment:
Can you somehow consolidate the isValid() and getValidationErrors() so that the check for what's valid timestamp is done only in one place? 

One way of doing that is to maybe provide isTooOld() or isWayInFuture() and use these methods in both isValid() and getValidationErrors()? That way we don't replicate the logic in 2 different places.